src/fundamental/macro-fundamental.h

```diff
@@ -11,6 +11,12 @@
 #include <stddef.h>
 #include <stdint.h>
 
+/* __FILE_NAME__ is gcc 12+ and clang 9+. Note that __builtin_strrchr() is always evaluated
+ * at compile time. */
+#ifndef __FILE_NAME__
```

Comment:
`__` prefix is special in C, it's a namespace owned by the language. Please don't squat it.